src/xdist/remote.py

```diff
@@ -116,6 +116,20 @@ def run_one_test(self, torun):
             "runtest_protocol_complete", item_index=self.item_index, duration=duration
         )
 
+    def pytest_collection_modifyitems(self, session, config, items):
+        # add the group name to nodeid as suffix if --dist=loadgroup
+        if config.getvalue("loadgroup"):
+            for item in items:
+                try:
+                    mark = item.get_closest_marker("xgroup")
+                except AttributeError:
+                    mark = item.get_marker("xgroup")
+
+                if mark:
+                    gname = mark.kwargs.get("name")
```

Comment:
Probably forcing to use `name` as keyword argument will be a bit surprising, we should also support positional argument (`@pytest.mark.xgroup("mygroup")`), otherwise this will get ignored.

```suggestion
                    gname = mark.args[0] if len(mark.args) > 0 else mark.kwargs.get("name")
```

Also not passing a group name should group to a default group, instead of not grouping at all?

Seems reasonable that all tests marked with `@pytest.mark.xgroup` would be grouped together as one group, and others marked with `@pytest.mark.xgroup("mygroup")` would be grouped in another group. What do you think?

Reply:
Thank you. 62e50d0